Dispatcher/UI/Homepage/HomepageViewController.swift

```diff
@@ -0,0 +1,145 @@
+    }
+    
+    private func configureActivityIndicator() {
+        activityIndicator.style = .large
+        activityIndicator.center = homepageView.center
+        homepageView.addSubview(activityIndicator)
+
+    }
+    
+    private func getArticles() {
+        self.activityIndicator.startAnimating()
+
+        homepageViewModel.getArticlesToDisplay(completionHandler: { errorMsg in
+            self.activityIndicator.stopAnimating()
+            if(errorMsg != nil) {
+                self.present(createErrorAlert(errorMsg!), animated: true, completion: nil)
+            } else {
+                DispatchQueue.main.async {
+                    self.articlesTableView.reloadData()
+                }
+            }
+        })
+    }
+    
+    @IBAction func searchButtonPressed(_ sender: UIButton) {
+        
+    }
+    
+    
+    @IBAction func notificationsButtonPressed(_ sender: UIButton) {
+        
+        
+    }
+    
+    @IBAction func unwindSegue( _ segue: UIStoryboardSegue) {
+    }
+    
+}
+
+extension HomepageViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return homepageViewModel.articlesAmount()
+        
+    }
+
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        
+        let currentArticle = homepageViewModel.getArticleByIndex(index: indexPath.row)
+        
+        if let cell = (tableView.dequeueReusableCell(withIdentifier: TableCellsIdentifiers.articleCellIdentifier, for: indexPath) as? ArticleCell) {
+            DispatchQueue.main.async {
+                cell.id = currentArticle.id
+                cell.autherLabel.text = currentArticle.author
+                cell.titleLabel.text = currentArticle.title
+                cell.subTitleLabel.text = currentArticle.summary
+                cell.tagLabel.text = currentArticle.topic.first
+                cell.dateLabel.text = formatDate(currentArticle.date)
+                cell.articleImage.image = loadImageFromUrl(currentArticle.imageURL)
+                
+                let numberOfTags = currentArticle.topic.count - 1
+                if(numberOfTags > 0) {
+                    cell.moreTagsLabel.text = "+ \(numberOfTags)"
+                } else {
+                    cell.moreTagsLabel.text = cell.tagLabel.text
+                    cell.tagLabel.isHidden = true
+                }
+                
+                cell.delegate = self
+            }
+            return cell
+        } else {
+            return UITableViewCell()
+        }
+    }
+    
+    func tableView(_ tableView: UITableView, willDisplay cell: UITableViewCell, forRowAt indexPath: IndexPath) {
+        if indexPath.row == (homepageViewModel.articlesToDisplay.count - 2) {
+            if (homepageViewModel.currentPage < homepageViewModel.totalResultsPages) {
+                getArticles()
```

Comment:
Eventually we reload the entire tableView just because we added a few more articles (possibly even just 1) to our existing data array (could already have even 100 items).
This is wasteful in resources.
Instead need to use insertRows to just add the new rows to the tableview and leave the existing rows untouched